community/cypher/cypher/src/main/scala/org/neo4j/cypher/ExecutionEngine.scala

```diff
@@ -136,8 +137,9 @@ class ExecutionEngine(graph: GraphDatabaseService, logProvider: LogProvider = Nu
     preParsedQueries.getOrElseUpdate(queryText, compiler.preParseQuery(queryText))
 
   @throws(classOf[SyntaxException])
-  protected def planQuery(queryText: String): (PreparedPlanExecution, TransactionInfo) =
-    using(compilationTracer.compileQuery(queryText)) { phaseTracer: CompilationPhaseTracer =>
+  protected def planQuery(queryText: String): (PreparedPlanExecution, TransactionInfo) = {
+    val phaseTracer = compilationTracer.compileQuery(queryText)
```

Comment:
This will make it so that the tracer no longer works. `using` is basically a try-with-resources, with the code looking like this no one will close the tracer. If this is a bottleneck I guess we should replace `using` with a straight up `try...finally`?